dev-packages/application-package/src/environment.ts

```diff
@@ -49,6 +49,28 @@ class ElectronEnv {
             && ((process as any).defaultApp || /node_modules[/]electron[/]/.test(process.execPath)); // tslint:disable-line:no-any
     }
 
+    /**
+     * Creates and return with a new environment object which always contains the `ELECTRON_RUN_AS_NODE: 1` property pair.
+     * This should be used to `spawn` and `fork` a new Node.js process from the Node.js shipped with Electron. Otherwise, a new Electron
+     * process will be spawned which [has side-effects](https://github.com/theia-ide/theia/issues/5385).
+     *
+     * If called from the backend and the `env` argument is not defined, it falls back to `process.env` such as Node.js behaves
+     * with the [`SpawnOptions`](https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options).
+     * If `env` is defined, it will be shallow-copied.
+     *
+     * Calling this function from the frontend does not make any sense, hence throw an error.
+     */
+    // tslint:disable-next-line:no-any
+    runAsNodeEnv(env?: any): any & { ELECTRON_RUN_AS_NODE: 1 } {
+        if (typeof process === 'undefined') {
+            throw new Error('Calling this function from the frontend does not make sense. Does it?');
```

Comment:
[minor] it's unlikely to happen, but should this error message be improved to display something more meaningful?

Reply:
👍 Sure, let's improve it. Do you have anything specific in mind?

Reply:
@kittaakos I don't have anything specific in mind, it just felt odd to me that we said it doesn't make sense to call the method from the frontend, but then said 'does it?' like we weren't sure. :see_no_evil: 

A generic error message stating the process is `undefined` may be sufficient.